jep/0000/README.adoc

```diff
@@ -0,0 +1,366 @@
+Integration server like link:https://ci.jenkins.io[https://ci.jenkins.io]. 
+
+The notion of continuous delivery of plugin releases has been discussed previously <<footnote-1,^(1)^>>.
+Considering that Jenkins is a system used to facilitate Continuous Delivery for many users, it makes 
+sense -- and builds credibility -- for the Jenkins developer community to adopt this same practice. 
+
+Having a centralized release system made available to plugin maintainers also provides additional 
+confidence that security best practices are being followed. 
+
+Continuous delivery from trusted CI is something which plugin maintainers can opt in to, but is 
+not required. If a plugin maintainer chooses to continue to follow their own path for releasing 
+versions of their code, they remain free to do so.
+
+== Reasoning
+
+=== What's with the version numbering?
+
+Jenkins plugin maintainers are already familiar with the way that Incrementals appends a commit 
+number, plus a SHA, to version numbers. These mechanically-generated version numbers offer 
+the ability to predictably sort them, so that external systems, such as the Jenkins update center, 
+can correctly publish the "newest" version. The addition of a merge SHA also allows for at-a-glance 
+feedback to tell people which commits went into the release.
+
+WARNING: We know the problem of consistent version numbering, and the resulting ability to 
+sort versions, is important. The scheme discussed here is only one of several possible 
+choices. We expect some _lively debate_ around this topic. But it's very important to ensure 
+that versions can be easily sorted by systems such as the Jenkins update center. Also under 
+consideration is defining a fixed number of fields in version numbers, and considering any 
+deviation from that a validation failure.
+
+=== Why YAML?
+YAML is becoming increasingly common in the Jenkins community, for many reasons. YAML is:
+
+* Human readable
+* In use by things like the Kubernetes plugin
+* Easily parsed by any number of publicly available libraries
+
+=== Webhooks and the Webhook Receiver
+Webhooks will be used to trigger the builds on the trusted CI server. The hooks themselves 
+will be configured on a per-repository basis. The reasons for choosing individual hooks, 
+as opposed to an organization wide hook, are:
+
+* Security of the build systems. Because this trusted CI server will reside on a non-public 
+network, there needs to be a proxy server of sorts, which can accept these webhooks and send 
+them, securely, to the trusted CI server.
+
+
+* Traceability. An organization-wide webhook would send those hooks to the receiver far more 
+often than is actually necessary. In the event that something goes wrong with this process - 
+the receiver goes down, there is a network outage, etc. - it will be easier for the Infra 
+team to triage problems if they're only looking at hooks from plugins which are intended 
+to be released via CD.
```

Comment:
But conversely you are adding complexity by needing to check whether each repo maintainer remembered to configure this webhook, and configured it correctly. Seems simpler to have an org hook and simply discard events from nonmatching plugins.